src/charm.py

```diff
@@ -76,7 +64,7 @@ def __init__(self, *args: typing.Any) -> None:
             strip_prefix=True,
         )
         self._observability = Observability(self)
-        self.framework.observe(self.on.leader_elected, self._on_leader_elected)
+        self._mjolnir = Mjolnir(self, charm_state=self._charm_state)
```

Comment:
Something else that comes to mind, should we consider naming this differently? From someone that doesn't know much about Synapse the `Mjolnir` name doesn't tell them that this is about moderation. Perhaps `MjolnirModeration` or something like it?

Reply:
What if I add a comment explaining what is Mjolnir? And more description to the class as well.
I'm afraid that changing it to MjolnirModeration might give the idea that it is something that will moderate "Mjolnir".

Reply:
comment would be nice

Reply:
Added :)